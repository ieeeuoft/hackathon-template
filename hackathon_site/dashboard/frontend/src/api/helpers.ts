import {
    Order,
    OrderInTable,
    OrderItemTableRow,
    ReturnedItem,
    ReturnOrderInTable,
} from "api/types";

export const formatDateTime = (dateTimeString: string): string => {
    const dateTime = new Date(dateTimeString);
    const options: Intl.DateTimeFormatOptions = {
        month: "long",
        day: "numeric",
        year: "numeric",
        hour: "numeric",
        minute: "numeric",
        hour12: true,
    };
    return dateTime.toLocaleString("en-US", options);
};
export const teamOrderListSerialization = (
    orders: Order[]
): {
    pendingOrders: OrderInTable[];
    checkedOutOrders: OrderInTable[];
    returnedOrders: ReturnOrderInTable[];
    hardwareIdsToFetch: number[];
} => {
    const pendingOrders: OrderInTable[] = [];
    const checkedOutOrders: OrderInTable[] = [];
    const returnedOrders: ReturnOrderInTable[] = [];
    const hardwareIdsToFetch: Record<number, number> = {};
    orders.forEach((order) => {
        if (order.status !== "Cancelled") {
            const hardwareItems: Record<number, OrderItemTableRow> = {};
            const hardwareRequested: Record<number, number> = {};
            const returnedItems: Record<string, ReturnedItem> = {};
            order.request.forEach(
                (hardware) =>
                    (hardwareRequested[hardware.id] = hardware.requested_quantity)
            );
            order.items.forEach(({ id, hardware_id, part_returned_health }) => {
                if (part_returned_health) {
                    const returnItemKey = `${hardware_id}-${part_returned_health}`;
                    if (returnedItems[returnItemKey])
                        returnedItems[returnItemKey].quantity += 1;
                    else {
                        const date = new Date(order.updated_at);
                        returnedItems[returnItemKey] = {
                            id,
                            quantity: 1,
                            part_returned_health,
                            hardware_id,
                            time: `${date.toLocaleTimeString()} (${date.toDateString()})`,
                        };
                    }
                } else {
                    if (hardwareItems[hardware_id])
                        hardwareItems[hardware_id].quantityGranted += 1;
                    else
                        hardwareItems[hardware_id] = {
                            id: hardware_id,
                            quantityGranted: 1,
                            quantityRequested: hardwareRequested[hardware_id],
                        };
                }
                hardwareIdsToFetch[hardware_id] = hardware_id;
            });
            const returnedHardware = Object.values(returnedItems);
            if (returnedHardware.length)
                returnedOrders.push({
                    id: order.id,
                    hardwareInOrder: returnedHardware,
                });

            const hardwareInTableRow = Object.values(hardwareItems);
            if (hardwareInTableRow.length > 0)
                (order.status === "Submitted" || order.status === "Ready for Pickup"
                    ? pendingOrders
                    : checkedOutOrders
                ).push({
                    id: order.id,
                    status: order.status,
                    hardwareInTableRow,
                    createdTime: order.created_at,
                    updatedTime: order.updated_at,
                });
        }
    });
    return {
        pendingOrders: [...pendingOrders].sort((a, b) => b.id - a.id),
        checkedOutOrders: [...checkedOutOrders].sort((a, b) => b.id - a.id),
        returnedOrders,
        hardwareIdsToFetch: Object.values(hardwareIdsToFetch),
    };
};

export const sortReturnedOrders = (
    order1: ReturnOrderInTable,
    order2: ReturnOrderInTable
): number => {
    const orderDate1 = order1.hardwareInOrder[0].time;
    const orderDate2 = order2.hardwareInOrder[0].time;

    const matchResult = orderDate1.match(
        /(\d{1,2}):(\d{2}):(\d{2}) (AM|PM) \((\w{3}) (\w{3}) (\d{2}) (\d{4})\)/
    );
    const matchResult2 = orderDate2.match(
        /(\d{1,2}):(\d{2}):(\d{2}) (AM|PM) \((\w{3}) (\w{3}) (\d{2}) (\d{4})\)/
    );

    if (matchResult && matchResult2) {
        // converting invalid date to a valid Date for first order to be compared
        const [, hours, minutes, seconds, meridiem, , month, day, year] = matchResult;
        const monthIndex = [
            "Jan",
            "Feb",
            "Mar",
            "Apr",
            "May",
            "Jun",
            "Jul",
            "Aug",
            "Sep",
            "Oct",
            "Nov",
            "Dec",
        ].indexOf(month);

        const date = new Date(
            Number(year),
            monthIndex,
            Number(day),
            Number(hours) + (meridiem === "PM" ? 12 : 0),
            Number(minutes),
            Number(seconds)
        );
        const formattedDate = date.toISOString();

        // converting invalid date to a valid Date for first order to be compared
        const [, hours2, minutes2, seconds2, meridiem2, , month2, day2, year2] =
            matchResult2;
        const monthIndex2 = [
            "Jan",
            "Feb",
            "Mar",
            "Apr",
            "May",
            "Jun",
            "Jul",
            "Aug",
            "Sep",
            "Oct",
            "Nov",
            "Dec",
        ].indexOf(month2);

        const date2 = new Date(
            Number(year2),
            monthIndex2,
            Number(day2),
            Number(hours2) + (meridiem2 === "PM" ? 12 : 0),
            Number(minutes2),
            Number(seconds2)
        );
        const formattedDate2 = date2.toISOString();

        return new Date(formattedDate2).valueOf() - new Date(formattedDate).valueOf();
    }
    return 0;
};

export const sortCheckedOutOrders = (
    order1: OrderInTable,
    order2: OrderInTable
): number => {
    return (
        new Date(order2.updatedTime).valueOf() - new Date(order1.updatedTime).valueOf()
    );
};

export const sortPendingOrders = (orders: OrderInTable[]): OrderInTable[] => {
    let ready_orders = [];
    let submitted_orders = [];
    for (let order of orders) {
        if (order.status === "Ready for Pickup") {
            ready_orders.push(order);
        } else {
            submitted_orders.push(order);
        }
    }
    ready_orders.sort((order1, order2) => {
        return (
            new Date(order1.updatedTime).valueOf() -
            new Date(order2.updatedTime).valueOf()
        );
    });

    submitted_orders.sort((order1, order2) => {
        return (
            new Date(order1.updatedTime).valueOf() -
            new Date(order2.updatedTime).valueOf()
        );
    });

    orders.splice(0, orders.length, ...submitted_orders, ...ready_orders);
    return orders;
};
