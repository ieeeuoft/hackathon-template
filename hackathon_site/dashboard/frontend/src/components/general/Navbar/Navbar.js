import React from "react";
import styles from "./Navbar.module.scss";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
// Images and logos
import DashboardIcon from "@material-ui/icons/Dashboard";
import LocalMallIcon from "@material-ui/icons/LocalMall";
import ListAlt from "@material-ui/icons/ListAlt";
import AccountBoxIcon from "@material-ui/icons/AccountBox";
import InsertChartOutlined from "@material-ui/icons/InsertChartOutlined";
import { ReactComponent as Inventory } from "assets/images/icons/Hardware.svg";
// Components
import Button from "@material-ui/core/Button";

const UnconnectedNavbar = ({ cartQuantity, pathname }) => (
    <nav className={styles.nav}>
        <div className={styles.navFlexDiv}>
            <Link to={"/"}>
                <Button
                    className={pathname === "/" ? styles.navActive : styles.navBtn}
                    aria-label="Dashboard"
                    startIcon={<DashboardIcon />}
                >
                    Dashboard
                </Button>
            </Link>
            <Link to={"/orders"}>
                <Button
                    className={
                        pathname === "/orders" ? styles.navActive : styles.navBtn
                    }
                    aria-label="Orders"
                    startIcon={<ListAlt />}
                >
                    Orders
                </Button>
            </Link>
            <Link to={"/teams"}>
                <Button
                    className={pathname === "/teams" ? styles.navActive : styles.navBtn}
                    aria-label="Teams"
                    startIcon={<AccountBoxIcon />}
                >
                    Teams
                </Button>
            </Link>
            <Link to={"/reports"}>
                <Button
                    className={
                        pathname === "/reports" ? styles.navActive : styles.navBtn
                    }
                    aria-label="Reports"
                    startIcon={<InsertChartOutlined />}
                >
                    Reports
                </Button>
            </Link>
            <Link to={"/inventory"}>
                <Button
                    className={
                        pathname === "/inventory" ? styles.navActive : styles.navBtn
                    }
                    aria-label="Inventory"
                    startIcon={<Inventory fill="currentColor" width="20px" />}
                >
                    Inventory
                </Button>
            </Link>
            <Link to={"/cart"}>
                <Button
                    className={pathname === "/cart" ? styles.navActive : styles.navBtn}
                    aria-label="Cart"
                    startIcon={<LocalMallIcon />}
                >
                    Cart ({cartQuantity})
                </Button>
            </Link>
        </div>
        <Button
            aria-label="Logout"
            className={`${styles.navBtn} ${styles.navBtnLogout}`}
        >
            <b>Logout</b>
        </Button>
    </nav>
);

const mapStateToProps = (state) => ({
    pathname: state.router.location.pathname,
});

const Navbar = connect(mapStateToProps)(UnconnectedNavbar);

export default Navbar;
