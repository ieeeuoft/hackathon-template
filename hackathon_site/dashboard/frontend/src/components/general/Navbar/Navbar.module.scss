@import "./../../../assets/abstracts/mixins";
@import "./../../../assets/abstracts/variables";

.nav {
    @include responsive(md-up) {
        &FlexDiv {
            display: inline-block;
        }
        &Btn,
        &IconBtn {
            color: color(white);
            .customIcon {
                width: 20px;
                fill: color(white);
            }
            &:hover {
                background-color: rgba(355, 355, 355, 0.2);
            }
        }
        &Btn {
            border-radius: 0;
            padding: 20px 15px;
        }
        &IconBtn {
            margin-left: 15px;
        }
    }

    @include responsive(md-down) {
        width: 100%;
        height: 100%;
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        &Btn {
            height: 50px;
            width: 100%;
            display: flex;
            justify-content: left;
            padding: 0px 120px 0px 15px;
            border-radius: 0;
            svg {
                margin-right: 30px;
            }
            &Logout {
                padding-left: 70px;
            }
        }
        .customIcon {
            width: 20px;
            fill: color(font);
        }
    }
}
