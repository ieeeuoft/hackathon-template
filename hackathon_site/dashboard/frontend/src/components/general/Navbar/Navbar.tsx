import React from "react";
import styles from "./Navbar.module.scss";
import { Link } from "react-router-dom";
import { connect, useSelector } from "react-redux";
// Images and logos
import DashboardIcon from "@material-ui/icons/Dashboard";
import LocalMallIcon from "@material-ui/icons/LocalMall";
import ListAlt from "@material-ui/icons/ListAlt";
import AccountBoxIcon from "@material-ui/icons/AccountBox";
import InsertChartOutlined from "@material-ui/icons/InsertChartOutlined";
import { ReactComponent as Inventory } from "assets/images/icons/Hardware.svg";
// Components
import Button from "@material-ui/core/Button";

import { logout, userTypeSelector } from "slices/users/userSlice";
import { RootState } from "slices/store";
import { cartTotalSelector } from "slices/hardware/cartSlice";

interface NavBarProps {
    logout: any;
    pathname: string;
}

export const UnconnectedNavbar = ({ logout, pathname }: NavBarProps) => {
    const cartQuantity = useSelector(cartTotalSelector);
    const userType = useSelector(userTypeSelector);

    const isParticipant = userType === "participant";
    const isAdmin = userType === "admin";
    const isParticipantOrAdmin = isParticipant || isAdmin;

    return (
        <nav className={styles.nav}>
            <div className={styles.navFlexDiv}>
                {isParticipantOrAdmin && (
                    <Link to={"/"}>
                        <Button
                            className={
                                pathname === "/" ? styles.navActive : styles.navBtn
                            }
                            aria-label="Dashboard"
                            startIcon={<DashboardIcon />}
                        >
                            Dashboard
                        </Button>
                    </Link>
                )}
                {isAdmin && (
                    <Link to={"/orders"}>
                        <Button
                            className={
                                pathname === "/orders"
                                    ? styles.navActive
                                    : styles.navBtn
                            }
                            aria-label="Orders"
                            startIcon={<ListAlt />}
                        >
                            Orders
                        </Button>
                    </Link>
                )}
                {isAdmin && (
                    <Link to={"/teams"}>
                        <Button
                            className={
                                pathname === "/teams" ? styles.navActive : styles.navBtn
                            }
                            aria-label="Teams"
                            startIcon={<AccountBoxIcon />}
                        >
                            Teams
                        </Button>
                    </Link>
                )}
                {isAdmin && (
                    <Link to={"/reports"}>
                        <Button
                            className={
                                pathname === "/reports"
                                    ? styles.navActive
                                    : styles.navBtn
                            }
                            aria-label="Reports"
                            startIcon={<InsertChartOutlined />}
                        >
                            Reports
                        </Button>
                    </Link>
                )}
                {isParticipantOrAdmin && (
                    <Link to={"/inventory"}>
                        <Button
                            className={
                                pathname === "/inventory"
                                    ? styles.navActive
                                    : styles.navBtn
                            }
                            aria-label="Inventory"
                            startIcon={<Inventory fill="currentColor" width="20px" />}
                        >
                            Inventory
                        </Button>
                    </Link>
                )}
                {isParticipant && (
                    <Link to={"/cart"}>
                        <Button
                            className={
                                pathname === "/cart" ? styles.navActive : styles.navBtn
                            }
                            aria-label="Cart"
                            startIcon={<LocalMallIcon />}
                        >
                            Cart ({cartQuantity})
                        </Button>
                    </Link>
                )}
            </div>
            <Button
                aria-label="Logout"
                className={`${styles.navBtn} ${styles.navBtnLogout}`}
                onClick={() => {
                    logout();
                }}
            >
                <b>Logout</b>
            </Button>
        </nav>
    );
};

const mapStateToProps = (state: RootState) => ({
    pathname: state.router.location.pathname,
});

const Navbar = connect(mapStateToProps, { logout })(UnconnectedNavbar);

export default Navbar;
