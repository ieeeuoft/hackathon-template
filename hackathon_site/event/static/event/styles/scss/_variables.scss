$colors: (
    font: #222,
    white: #fff,
    primary: #1976d2,
    secondary: #ffa000,
    light: #b7dbff,
    dark: #002f5e,
    backgroundColor: #ddeeff,
    error: #b00020,
    success: #0b890b,
    primaryOnHover: #08529c,
    secondaryOnHover: #ce8100,
    errorOnHover: #870000,
    successOnHover: #1d7521,
);

$fonts: (
    body: "Nunito",
    header: "Roboto",
);

$font-sizes: (
    xl: 20px,
    lg: 18px,
    md: 16px,
    sm: 14px,
    xs: 12px,
);

@function color($color-name) {
    @return map-get($colors, $color-name); //(name of map, key)
}

@function font($fonts-name) {
    @return map-get($fonts, $fonts-name); //(name of map, key)
}

@function size($size) {
    @return map-get($font-sizes, $size); //(name of map, key)
}
