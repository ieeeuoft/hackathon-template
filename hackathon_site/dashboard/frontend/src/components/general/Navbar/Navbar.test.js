import React from "react";
import { render } from "@testing-library/react";
import Navbar from "./Navbar";
import { styles } from "./Navbar.module.scss";
import { withStoreAndRouter } from "testing/helpers";
import { cartQuantity } from "testing/mockData";

describe("<Navbar />", () => {
    const pagesAndPaths = [
        ["Dashboard", "/"],
        ["Orders", "/orders"],
        ["Teams", "/teams"],
        ["Reports", "/reports"],
        ["Inventory", "/inventory"],
        [`Cart (${cartQuantity})`, "/cart"],
    ];

    pagesAndPaths.map(([label, path]) => {
        it(`Adds the active class to ${label} when on ${path}`, () => {
            const { getByText, container } = render(
                withStoreAndRouter(
                    <Navbar cartQuantity={cartQuantity} pathname={path} />
                )
            );

            expect(getByText(label).closest("button").className).toMatch(
                new RegExp(styles.navActive)
            );

            expect(container.querySelectorAll(".navActive").length).toBe(1);
        });
    });
});
