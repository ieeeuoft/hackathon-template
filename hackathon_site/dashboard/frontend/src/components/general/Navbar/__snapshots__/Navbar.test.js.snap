// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly when all icons appear 1`] = `
<DocumentFragment>
  <nav
    class="nav"
  >
    <div
      class="navFlexDiv"
    >
      <a
        href="/"
      >
        <button
          aria-label="Dashboard"
          class="MuiButtonBase-root MuiButton-root MuiButton-text navBtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                />
              </svg>
            </span>
            Dashboard
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        href="/orders"
      >
        <button
          aria-label="Orders"
          class="MuiButtonBase-root MuiButton-root MuiButton-text navBtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m1.1-2H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9zM11 7h6v2h-6V7zm0 4h6v2h-6v-2zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7z"
                />
              </svg>
            </span>
            Orders
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        href="/teams"
      >
        <button
          aria-label="Teams"
          class="MuiButtonBase-root MuiButton-root MuiButton-text navBtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 5v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H5c-1.11 0-2 .9-2 2zm12 4c0 1.66-1.34 3-3 3s-3-1.34-3-3 1.34-3 3-3 3 1.34 3 3zm-9 8c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1H6v-1z"
                />
              </svg>
            </span>
            Teams
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        href="/reports"
      >
        <button
          aria-label="Reports"
          class="MuiButtonBase-root MuiButton-root MuiButton-text navBtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4zm2.5 2.1h-15V5h15v14.1zm0-16.1h-15c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h15c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            Reports
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        href="/inventory"
      >
        <button
          aria-label="Inventory"
          class="MuiButtonBase-root MuiButton-root MuiButton-text navBtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                class="customIcon"
              >
                Hardware.svg
              </svg>
            </span>
            Inventory
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        href="/cart"
      >
        <button
          aria-label="Cart"
          class="MuiButtonBase-root MuiButton-root MuiButton-text navBtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6h-2c0-2.76-2.24-5-5-5S7 3.24 7 6H5c-1.1 0-1.99.9-1.99 2L3 20c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm-7-3c1.66 0 3 1.34 3 3H9c0-1.66 1.34-3 3-3zm0 10c-2.76 0-5-2.24-5-5h2c0 1.66 1.34 3 3 3s3-1.34 3-3h2c0 2.76-2.24 5-5 5z"
                />
              </svg>
            </span>
            Cart (1)
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
    <button
      aria-label="Logout"
      class="MuiButtonBase-root MuiButton-root MuiButton-text navBtn navBtnLogout"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <b>
          Logout
        </b>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </nav>
</DocumentFragment>
`;

exports[`<Navbar /> renders correctly when all icons appear but no badge on notification iconon mobile 1`] = `
<DocumentFragment>
  <nav
    class="nav"
  >
    <div
      class="navFlexDiv"
    >
      <a
        href="/"
      >
        <button
          aria-label="Dashboard"
          class="MuiButtonBase-root MuiButton-root MuiButton-text navBtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                />
              </svg>
            </span>
            Dashboard
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        href="/orders"
      >
        <button
          aria-label="Orders"
          class="MuiButtonBase-root MuiButton-root MuiButton-text navBtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m1.1-2H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9zM11 7h6v2h-6V7zm0 4h6v2h-6v-2zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7z"
                />
              </svg>
            </span>
            Orders
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        href="/teams"
      >
        <button
          aria-label="Teams"
          class="MuiButtonBase-root MuiButton-root MuiButton-text navBtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 5v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H5c-1.11 0-2 .9-2 2zm12 4c0 1.66-1.34 3-3 3s-3-1.34-3-3 1.34-3 3-3 3 1.34 3 3zm-9 8c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1H6v-1z"
                />
              </svg>
            </span>
            Teams
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        href="/reports"
      >
        <button
          aria-label="Reports"
          class="MuiButtonBase-root MuiButton-root MuiButton-text navBtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4zm2.5 2.1h-15V5h15v14.1zm0-16.1h-15c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h15c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            Reports
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        href="/inventory"
      >
        <button
          aria-label="Inventory"
          class="MuiButtonBase-root MuiButton-root MuiButton-text navBtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                class="customIcon"
              >
                Hardware.svg
              </svg>
            </span>
            Inventory
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        href="/cart"
      >
        <button
          aria-label="Cart"
          class="MuiButtonBase-root MuiButton-root MuiButton-text navBtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6h-2c0-2.76-2.24-5-5-5S7 3.24 7 6H5c-1.1 0-1.99.9-1.99 2L3 20c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm-7-3c1.66 0 3 1.34 3 3H9c0-1.66 1.34-3 3-3zm0 10c-2.76 0-5-2.24-5-5h2c0 1.66 1.34 3 3 3s3-1.34 3-3h2c0 2.76-2.24 5-5 5z"
                />
              </svg>
            </span>
            Cart (0)
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
    <button
      aria-label="Logout"
      class="MuiButtonBase-root MuiButton-root MuiButton-text navBtn navBtnLogout"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <b>
          Logout
        </b>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </nav>
</DocumentFragment>
`;
